pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -591,71 +591,65 @@ def specs(self) -> str:
         """Text summarizing the specifications of the device."""
         return self._specs(for_docs=False)
 
-    def _specs(self, for_docs: bool = False) -> str:
-        def yes_no_fn(cond: bool) -> str:
-            return "Yes" if cond else "No"
+    def _param_yes_no(self, param: Any) -> str:
+        return "Yes" if param is True else "No"
+
+    def _param_check_none(self, param: Any) -> Callable[[str], str]:
+        def empty_str_if_none(line: str) -> str:
+            if param is None:
+                return ""
+            else:
+                return line.format(param)
+
+        return empty_str_if_none
 
-        lines = [
+    def _register_lines(self) -> str:
+
+        register_lines = [
             "\nRegister parameters:",
             f" - Dimensions: {self.dimensions}D",
             f" - Rydberg level: {self.rydberg_level}",
-            f" - Maximum number of atoms: {self.max_atom_num}",
-            f" - Maximum distance from origin: {self.max_radial_distance} μm",
-            (
-                " - Minimum distance between neighbouring atoms: "
-                f"{self.min_atom_distance} μm"
+            self._param_check_none(self.max_atom_num)(
+                " - Maximum number of atoms: {}"
             ),
-            f" - Maximum layout filling fraction: {self.max_layout_filling}",
-            f" - SLM Mask: {yes_no_fn(self.supports_slm_mask)}",
+            self._param_check_none(self.max_radial_distance)(
+                " - Maximum distance from origin: {} µm"
+            ),
+            " - Minimum distance between neighbouring atoms: "
+            + f"{self.min_atom_distance} μm",
+            f" - SLM Mask: {self._param_yes_no(self.supports_slm_mask)}",
         ]
 
-        if self.max_sequence_duration is not None:
-            lines.append(
-                " - Maximum sequence duration: "
-                f"{self.max_sequence_duration} ns"
-            )
+        return "\n".join(line for line in register_lines if line != "")
```

Comment:
Can we return `list[str]` instead? This (and the other similar methods) could change to
```suggestion
        return [line for line in register_lines if line != ""]
```